engine/src/test/java/io/camunda/zeebe/engine/processing/job/ActivateJobsTest.java

```diff
@@ -455,6 +455,36 @@ public void shouldActivateJobUpToMaxMessageSize() {
         .isEqualTo(Map.of("foo", "x".repeat(variablesSize), "bar", "x".repeat(variablesSize)));
   }
 
+  // regression test for https://github.com/camunda/zeebe/issues/10308
+  @Test
+  public void shouldNotActivateJobWithNoRemainingRetries() {
```

Comment:
:x: Additionally, we should add a similar test case to the unit test for the state `JobStateTest`.

We need a similar test case as [shouldFailJobWithNoRetriesLeft()](https://github.com/camunda/zeebe/blob/main/engine/src/test/java/io/camunda/zeebe/engine/state/instance/JobStateTest.java#L428-L445) but without activation.